privilege/privileges/privileges.go

```diff
@@ -240,7 +244,10 @@ func (p *UserPrivileges) GetAuthPlugin(user, host string) (string, error) {
 	if record == nil {
 		return "", errors.New("Failed to get user record")
 	}
-	if len(record.AuthenticationString) == 0 {
+	// zero-length auth string means no password for native and caching_sha2 auth.
+	// but for auth_socket it means there should be a 1-to-1 mapping between the TiDB user
+	// and the OS user.
+	if (len(record.AuthenticationString) == 0) && (record.AuthPlugin != mysql.AuthSocket) {
```

Comment:
I think this is slightly more readable:

```suggestion
	if record.AuthenticationString == "" && record.AuthPlugin != mysql.AuthSocket {
```

I seem to remember a linter that prefered `str = ""` instead of `len(str) == 0`, but I can't find it now. From googling it looks like the `len` func is optimized out anyway.